tests/bsim/bluetooth/mesh/src/test_advertiser.c

```diff
@@ -580,6 +580,189 @@ static void test_rx_random_order(void)
+
+	ASSERT_EQUAL(err, 0);
+
+	if (adv_data->suspend) {
+		/* When suspending, the end callback will be called only for the first adv, because
+		 * it was already scheduled.
+		 */
+		ASSERT_EQUAL(adv_data->i, 0);
+	} else {
+		ASSERT_EQUAL(err, 0);
+
+		if (adv_data->i == CONFIG_BT_MESH_ADV_BUF_COUNT - 1) {
+			k_sem_give(&adv_sent_sem);
+		}
+	}
+}
+
+static void adv_send_start(uint16_t duration, int err, void *cb_data)
+{
+	struct adv_suspend_ctx *adv_data = cb_data;
+
+	LOG_DBG("start(): err (%d), suspend (%d), i (%d)", err, adv_data->suspend, adv_data->i);
+
+	if (adv_data->suspend) {
+		if (adv_data->i == 0) {
+			ASSERT_EQUAL(err, 0);
+			k_work_submit(&adv_disable_work);
+		} else {
+			/* For the advs that were pushed to the mesh advertiser by calling
+			 * `bt_mesh_adv_send` function but not sent to the host, the start callback
+			 * shall be called with -ENODEV.
+			 */
+			ASSERT_EQUAL(err, -ENODEV);
+		}
+
+		if (adv_data->i == CONFIG_BT_MESH_ADV_BUF_COUNT - 1) {
+			k_sem_give(&adv_suspended_sem);
+		}
+	} else {
+		ASSERT_EQUAL(err, 0);
+	}
+}
+
+static void adv_create_and_send(bool suspend, uint8_t first_byte, struct adv_suspend_ctx *adv_data)
+{
+	struct bt_mesh_adv *advs[CONFIG_BT_MESH_ADV_BUF_COUNT];
+	static const struct bt_mesh_send_cb send_cb = {
+		.start = adv_send_start,
+		.end = adv_send_end,
+	};
+
+	for (int i = 0; i < CONFIG_BT_MESH_ADV_BUF_COUNT; i++) {
+		adv_data[i].suspend = suspend;
+		adv_data[i].i = i;
+
+		advs[i] = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+					    BT_MESH_TRANSMIT(2, 20), K_NO_WAIT);
+		ASSERT_FALSE_MSG(!advs[i], "Out of advs\n");
+
+		net_buf_simple_add_u8(&advs[i]->b, first_byte);
+		net_buf_simple_add_u8(&advs[i]->b, i);
+	}
+
+	for (int i = 0; i < CONFIG_BT_MESH_ADV_BUF_COUNT; i++) {
+		bt_mesh_adv_send(advs[i], &send_cb, &adv_data[i]);
+		bt_mesh_adv_unref(advs[i]);
+	}
+}
+
+static void test_tx_disable(void)
+{
+	struct adv_suspend_ctx adv_data[CONFIG_BT_MESH_ADV_BUF_COUNT];
+	struct bt_mesh_adv *extra_adv;
+	int err;
+
+	bt_init();
+	adv_init();
+
+	/* Fill up the adv pool and suspend the advertiser in the first start callback call. */
+	adv_create_and_send(true, 0xAA, adv_data);
+
+	err = k_sem_take(&adv_suspended_sem, K_SECONDS(10));
+	ASSERT_OK_MSG(err, "Not all advs were sent");
+
+	extra_adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+				       BT_MESH_TRANSMIT(2, 20), K_NO_WAIT);
+	ASSERT_TRUE_MSG(!extra_adv, "Created adv while suspended");
+
+	adv_resume();
+
+	/* Fill up the adv pool and suspend the advertiser and let it send all advs. */
+	adv_create_and_send(false, 0xBB, adv_data);
+
+	err = k_sem_take(&adv_sent_sem, K_SECONDS(10));
+	ASSERT_OK_MSG(err, "Not all advs were sent");
+
+	PASS();
+}
+
+static void suspended_adv_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				  struct net_buf_simple *buf)
+{
+	uint8_t length;
+	uint8_t type;
+	uint8_t pdu;
+
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_NONCONN_IND) {
```

Comment:
Shouldn't it be the test failure since mesh was suspended?

Reply:
Other messages will be received

Reply:
Hm... whether are any proxy or pb-gatt messages sent in parallel to this test scenario?

Reply:
Only non connectable non-scannable undirected advertisements are sent in this test.

Reply:
Yep, that's right. It is what I see in PR. There shouldn't be any others, especially after advertiser suspending. Isn't receiving of connectable messages the misbehavior here?

Reply:
Why do we need to check this here? I don't see where I enable proxy advertisements.

Reply:
Following this logic, the condition is not required at all since there is no enabled proxy advertisements.